MetaMask/metamask-mobile
# fix: display hash as hex in personal signature confirmation

## **Description**

When displaying a personal sign confirmation, the extension displays data with a length of 32 bytes as the original hexadecimal string, rather than the utf8 equivalent. This is likely to acommodate cases where a hash is to be signed rather than a human readable message.

Mobile always displays the utf8 equivalent which results in invalid characters if a hash was provided.

See extension code [here](https://github.com/MetaMask/metamask-extension/blob/develop/ui/components/app/signature-request-original/signature-request-original.component.js#L94).

## **Related issues**

Fixes:

## **Manual testing steps**

1. Perform `personal_sign` request with the first argument (data) as a 32 byte hexadecimal string.
2. Ensure same string is visible in the confirmation.
3. Perform `personal_sign` with any other buffer size.
4. Ensure equivalent UTF8 is displayed.

## **Screenshots/Recordings**

### **Before**

<img width="180" alt="before" src="https://github.com/MetaMask/metamask-mobile/assets/19745142/045c4efe-f590-4f89-bf62-72d3533140bd">

### **After**

<img width="180" alt="after" src="https://github.com/MetaMask/metamask-mobile/assets/19745142/31e15447-dbef-4b59-82cc-e04f4242de87">

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
